tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -479,55 +482,73 @@ object AutoTuner extends Logging {
   val DEFAULT_WORKER_INFO: String = "."
   val SUPPORTED_SIZE_UNITS: Seq[String] = Seq("b", "k", "m", "g", "t", "p")
 
+  def loadSystemProperties(fileInput: String): Option[java.util.Map[String, Any]] = {
+    val filePath = new Path(fileInput)
+    val fs = Some(FileSystem.get(filePath.toUri, new Configuration()))
```

Comment:
why do we force to be Some and then do foreach on it?  If we know it exists and is valid we should just use it.

Reply:
Quick question just for sake of learning. 
How do we know it exists and is valid? `fileInput` is a user argument which might not be valid (e.g., `XYZ://file-Path`).
I see that in this piece of code, using `forEach` avoid throwing IOException but `parseSystemInfo` catches `Exception` anyway.

Reply:
right but what you have here is forcing it to be Some so it will always go into the foreach.  If the FileSystem.get was to return null on invalid path then you should want to do something like Option(Filesystem.get...).

```
scala> Option(null)
res3: Option[Null] = None
```

In this case though FileSystem.get I don't think returns null and just throws if something bad happens.  But if the path is just invalid I think it will still give you a filesystem and then when you go to read it will fail.  So I think you can just put val fs = FileSystem.get, get rid of the foreach code and try/catch the opening.